File: app/src/test/java/bj/discogsbrowser/artistreleases/ArtistReleasesControllerTest.java
package bj.discogsbrowser.artistreleases;

import android.content.Context;

import com.airbnb.epoxy.EpoxyModel;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.annotation.Config;

import java.util.List;

import bj.discogsbrowser.utils.ImageViewAnimator;

import static android.os.Build.VERSION_CODES.LOLLIPOP;
import static junit.framework.Assert.assertEquals;

/**
 * Created by Josh Laird on 09/05/2017.
 * <p>
 * Exact mirror of @{@link bj.discogsbrowser.singlelist.SingleListControllerTest}.
 */
@RunWith(RobolectricTestRunner.class)
@Config(sdk = LOLLIPOP, manifest = Config.NONE)
public class ArtistReleasesControllerTest
{
    private ArtistReleasesController controller;
    @Mock Context context;
    @Mock ArtistReleasesContract.View view;
    @Mock ImageViewAnimator imageViewAnimator;

    @Before
    public void setUp()
    {
        MockitoAnnotations.initMocks(this);
        controller = new ArtistReleasesController(context, view, imageViewAnimator);
        controller.requestModelBuild();
    }

    @Test
    public void initialState_Loading()
    {
        List<EpoxyModel<?>> copyOfModels = controller.getAdapter().getCopyOfModels();
        assertEquals(copyOfModels.get(0).getClass().getSimpleName(), "SmallEmptySpaceModel_");
        assertEquals(copyOfModels.get(1).getClass().getSimpleName(), "LoadingModel_");
        assertEquals(copyOfModels.get(2).getClass().getSimpleName(), "SmallEmptySpaceModel_");
        assertEquals(copyOfModels.size(), 3);
    }
}

File: app/src/test/java/bj/vinylbrowser/artistreleases/ArtistReleasesControllerTest.kt
package bj.vinylbrowser.artistreleases

import android.content.Context
import android.os.Build
import bj.vinylbrowser.utils.ImageViewAnimator
import com.nhaarman.mockito_kotlin.mock
import org.junit.Assert.assertEquals
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config

/**
 * Created by Josh Laird on 21/05/2017.
 */
@RunWith(RobolectricTestRunner::class)
@Config(sdk = intArrayOf(Build.VERSION_CODES.LOLLIPOP), manifest = Config.NONE)
class ArtistReleasesControllerTest {
    lateinit var controller: ArtistReleasesController
    val context: Context = mock()
    val view: ArtistReleasesContract.View = mock()
    val imageViewAnimator: ImageViewAnimator = mock()

    @Before
    fun setUp() {
        controller = ArtistReleasesController(context, view, imageViewAnimator)
        controller.requestModelBuild()
    }

    @Test
    fun initialState_Loading() {
        val copyOfModels = controller.adapter.copyOfModels
        assertEquals(copyOfModels[0].javaClass.simpleName, "SmallEmptySpaceModel_")
        assertEquals(copyOfModels[1].javaClass.simpleName, "LoadingModel_")
        assertEquals(copyOfModels[2].javaClass.simpleName, "SmallEmptySpaceModel_")
        assertEquals(copyOfModels.size, 3)
    }
}